bracket-push/bracket_push_test.spec.js

```diff
@@ -21,6 +21,10 @@ describe('bracket push', function() {
     expect(bracket('{[]}')).toEqual(true);
   });
 
+  xit('rejects brackets that are properly balanced but improperly nested', function() {
+    expect(bracket('{[}]')).toEqual(false);
```

Comment:
Reasoning for this being false: https://github.com/exercism/x-common/blob/master/bracket-push.md says **nested correctly**. Please sanity check on whether you agree with this interpretation.

If there is disagreement, we should at least make sure we reach the same conclusions as xgo (see linked PR for that)